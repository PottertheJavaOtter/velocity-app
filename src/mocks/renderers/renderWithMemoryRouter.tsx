// @ts-nocheck
import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router-dom";
import { History, LocationDescriptor } from "history";

const renderWithMemoryRouter = (
  ui: React.ReactNode,
  initialEntries: LocationDescriptor[] = ["/my/initial-route"]
) => {
  let historyInTest: History;
  return {
    ...render(
      <MemoryRouter initialEntries={initialEntries}>
        {ui}
        <Route
          path="*"
          render={({ history }) => {
            historyInTest = history;
            return null;
          }}
        />
      </MemoryRouter>),
    history: historyInTest
  };
};

export default renderWithMemoryRouter;
