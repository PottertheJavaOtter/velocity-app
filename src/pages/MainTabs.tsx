import React from "react";
import {
  IonTabs,
  IonRouterOutlet,
  IonTabBar,
  IonTabButton,
  IonIcon,
  IonLabel,
} from "@ionic/react";
import { Route, Redirect } from "react-router";
import {
  cashOutline,
  fileTrayFullOutline,
  homeOutline,
  calendarOutline,
} from "ionicons/icons";
import SchedulePage from "./SchedulePage";
import HomePage from "./HomePage/HomePage";
import Income from "./Income/Income";
import SessionDetail from "./SessionDetail";
import Portfolio from "./Portfolio/Portfolio";

interface MainTabsProps { }

const MainTabs: React.FC<MainTabsProps> = () => {
  return (
    <IonTabs>
      <IonRouterOutlet>
        <Route
          path="/tabs/home"
          render={() => <HomePage />}
          exact={true}
        />
        <Route
          path="/tabs/schedule"
          render={() => <SchedulePage />}
          exact={true}
        />
        <Route
          path="/tabs/income"
          render={() => <Income />}
          exact={true}
        />
        <Route path="/tabs/schedule/:id" component={SessionDetail} />
        <Route
          path="/tabs/porfolio"
          render={() => <Portfolio />}
          exact={true}
        />
        <Route
          path="/tabs"
          exact={true}
        >
          <Redirect exact to="/tabs/home" />
        </Route>
      </IonRouterOutlet>
      <IonTabBar slot="bottom">
        <IonTabButton tab="home" href="/tabs/home">
          <IonIcon icon={homeOutline} />
          <IonLabel>Home</IonLabel>
        </IonTabButton>
        <IonTabButton tab="porfolio" href="/tabs/porfolio">
          <IonIcon icon={fileTrayFullOutline} />
          <IonLabel>Porfolio</IonLabel>
        </IonTabButton>
        <IonTabButton tab="income" href="/tabs/income">
          <IonIcon icon={cashOutline} />
          <IonLabel>Income</IonLabel>
        </IonTabButton>
        <IonTabButton tab="schedule" href="/tabs/schedule">
          <IonIcon icon={calendarOutline} />
          <IonLabel>Payout Schedule</IonLabel>
        </IonTabButton>
      </IonTabBar>
    </IonTabs>
  );
};

export default MainTabs;
