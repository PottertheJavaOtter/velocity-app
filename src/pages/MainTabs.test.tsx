import React from 'react';
import renderWithMemoryRouter from '../mocks/renderers/renderWithMemoryRouter';
import MainTabs from './MainTabs';
import { screen } from '@testing-library/dom';

jest.mock("./SchedulePage", () => {
  return () => <div>On the Schedule Page</div>
});
jest.mock("./HomePage/HomePage", () => {
  return () => <div>On the Home Page</div>
})
jest.mock("./Income/Income", () => {
  return () => <div>On the Income Page</div>
})
jest.mock("./SessionDetail", () => {
  return () => <div>On the Session Page</div>
})
jest.mock("./Portfolio/Portfolio", () => {
  return () => <div>On the Portfolio Page</div>
})

describe("MainTabs", () => {
  describe('default route', () => {
    it('should take you to Home Page', () => {
      const { container, history } = renderWithMemoryRouter(<MainTabs />, ["/tabs"]);
      expect(screen.getByText("On the Home Page")).toBeInTheDocument();
      expect(history.location.pathname).toContain("/tabs/home")
      expect(container).toMatchSnapshot();
    })
  })

  describe('/home', () => {
    it('should take you to Home Page', () => {
      renderWithMemoryRouter(<MainTabs />, ["/tabs/home"]);
      expect(screen.getByText("On the Home Page")).toBeInTheDocument();
    })
  })

  describe('/schedule', () => {
    it('should take you to Schedule Page', () => {
      renderWithMemoryRouter(<MainTabs />, ["/tabs/schedule"]);
      expect(screen.getByText("On the Schedule Page")).toBeInTheDocument();
    })

    describe("/:id", () => {
      it('should take you to Session Page', () => {
        renderWithMemoryRouter(<MainTabs />, ["/tabs/schedule/some-id"]);
        expect(screen.getByText("On the Session Page")).toBeInTheDocument();
      })
    })
  })

  describe("isLogged in", () => {
    describe('/income', () => {
      it('should take you to Income Page', () => {
        renderWithMemoryRouter(<MainTabs isLoggedin />, ["/tabs/income"]);
        expect(screen.getByText("On the Income Page")).toBeInTheDocument();
      })
    })

    describe('/porfolio', () => {
      it('should take you to Home Page', () => {
        renderWithMemoryRouter(<MainTabs isLoggedin />, ["/tabs/porfolio"]);
        expect(screen.getByText("On the Portfolio Page")).toBeInTheDocument();
      })
    })
  })
});
