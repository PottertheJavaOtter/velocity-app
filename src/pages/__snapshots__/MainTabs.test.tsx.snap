// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTabs /home should take you to Home Page 1`] = `
<div>
  <div
    class="ion-tabs"
    style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
  >
    <div
      class="tabs-inner"
      style="position: relative; flex: 1; contain: layout size style;"
    >
      <ion-router-outlet>
        <div>
          On the Home Page
        </div>
      </ion-router-outlet>
    </div>
    <ion-tab-bar
      selected-tab="home"
      slot="bottom"
    >
      <ion-tab-button
        href="/"
        tab="home"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M80 212v236a16 16 0 0016 16h96V328a24 24 0 0124-24h80a24 24 0 0124 24v136h96a16 16 0 0016-16V212' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 256L266.89 52c-5-5.28-16.69-5.34-21.78 0L32 256M400 179V64h-48v69' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Home
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/porfolio"
        tab="porfolio"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M384 80H128c-26 0-43 14-48 40L48 272v112a48.14 48.14 0 0048 48h320a48.14 48.14 0 0048-48V272l-32-152c-5-27-23-40-48-40z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M48 272h144M320 272h144M192 272a64 64 0 00128 0M144 144h224M128 208h256' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Porfolio
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/income"
        tab="income"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='32' y='80' width='448' height='256' rx='16' ry='16' transform='rotate(180 256 208)' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M64 384h384M96 432h320' class='ionicon-fill-none ionicon-stroke-width'/><circle cx='256' cy='208' r='80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 160a80 80 0 01-80-80M32 160a80 80 0 0080-80M480 256a80 80 0 00-80 80M32 256a80 80 0 0180 80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Income
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/schedule"
        tab="schedule"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='48' y='80' width='416' height='384' rx='48' ry='48' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M397.82 80H114.18C77.69 80 48 110.15 48 147.2V208h16c0-16 16-32 32-32h320c16 0 32 16 32 32h16v-60.8c0-37.05-29.69-67.2-66.18-67.2z'/><circle cx='296' cy='232' r='24'/><circle cx='376' cy='232' r='24'/><circle cx='296' cy='312' r='24'/><circle cx='376' cy='312' r='24'/><circle cx='136' cy='312' r='24'/><circle cx='216' cy='312' r='24'/><circle cx='136' cy='392' r='24'/><circle cx='216' cy='392' r='24'/><circle cx='296' cy='392' r='24'/><path stroke-linecap='round' stroke-linejoin='round' d='M128 48v32M384 48v32' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Payout Schedule
        </ion-label>
      </ion-tab-button>
    </ion-tab-bar>
  </div>
</div>
`;

exports[`MainTabs /home should take you to Home Page 2`] = `
<div>
  <div
    class="ion-tabs"
    style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
  >
    <div
      class="tabs-inner"
      style="position: relative; flex: 1; contain: layout size style;"
    >
      <ion-router-outlet>
        <div>
          On the Home Page
        </div>
      </ion-router-outlet>
    </div>
    <ion-tab-bar
      selected-tab="home"
      slot="bottom"
    >
      <ion-tab-button
        href="/"
        tab="home"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M80 212v236a16 16 0 0016 16h96V328a24 24 0 0124-24h80a24 24 0 0124 24v136h96a16 16 0 0016-16V212' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 256L266.89 52c-5-5.28-16.69-5.34-21.78 0L32 256M400 179V64h-48v69' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Home
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/porfolio"
        tab="porfolio"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M384 80H128c-26 0-43 14-48 40L48 272v112a48.14 48.14 0 0048 48h320a48.14 48.14 0 0048-48V272l-32-152c-5-27-23-40-48-40z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M48 272h144M320 272h144M192 272a64 64 0 00128 0M144 144h224M128 208h256' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Porfolio
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/income"
        tab="income"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='32' y='80' width='448' height='256' rx='16' ry='16' transform='rotate(180 256 208)' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M64 384h384M96 432h320' class='ionicon-fill-none ionicon-stroke-width'/><circle cx='256' cy='208' r='80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 160a80 80 0 01-80-80M32 160a80 80 0 0080-80M480 256a80 80 0 00-80 80M32 256a80 80 0 0180 80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Income
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/schedule"
        tab="schedule"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='48' y='80' width='416' height='384' rx='48' ry='48' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M397.82 80H114.18C77.69 80 48 110.15 48 147.2V208h16c0-16 16-32 32-32h320c16 0 32 16 32 32h16v-60.8c0-37.05-29.69-67.2-66.18-67.2z'/><circle cx='296' cy='232' r='24'/><circle cx='376' cy='232' r='24'/><circle cx='296' cy='312' r='24'/><circle cx='376' cy='312' r='24'/><circle cx='136' cy='312' r='24'/><circle cx='216' cy='312' r='24'/><circle cx='136' cy='392' r='24'/><circle cx='216' cy='392' r='24'/><circle cx='296' cy='392' r='24'/><path stroke-linecap='round' stroke-linejoin='round' d='M128 48v32M384 48v32' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Payout Schedule
        </ion-label>
      </ion-tab-button>
    </ion-tab-bar>
  </div>
</div>
`;

exports[`MainTabs /home should take you to home 1`] = `
<div>
  <div
    class="ion-tabs"
    style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
  >
    <div
      class="tabs-inner"
      style="position: relative; flex: 1; contain: layout size style;"
    >
      <ion-router-outlet>
        <div>
          On the Home Page
        </div>
      </ion-router-outlet>
    </div>
    <ion-tab-bar
      selected-tab="home"
      slot="bottom"
    >
      <ion-tab-button
        href="/"
        tab="home"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M80 212v236a16 16 0 0016 16h96V328a24 24 0 0124-24h80a24 24 0 0124 24v136h96a16 16 0 0016-16V212' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 256L266.89 52c-5-5.28-16.69-5.34-21.78 0L32 256M400 179V64h-48v69' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Home
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/porfolio"
        tab="porfolio"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M384 80H128c-26 0-43 14-48 40L48 272v112a48.14 48.14 0 0048 48h320a48.14 48.14 0 0048-48V272l-32-152c-5-27-23-40-48-40z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M48 272h144M320 272h144M192 272a64 64 0 00128 0M144 144h224M128 208h256' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Porfolio
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/income"
        tab="income"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='32' y='80' width='448' height='256' rx='16' ry='16' transform='rotate(180 256 208)' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M64 384h384M96 432h320' class='ionicon-fill-none ionicon-stroke-width'/><circle cx='256' cy='208' r='80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 160a80 80 0 01-80-80M32 160a80 80 0 0080-80M480 256a80 80 0 00-80 80M32 256a80 80 0 0180 80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Income
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/schedule"
        tab="schedule"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='48' y='80' width='416' height='384' rx='48' ry='48' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M397.82 80H114.18C77.69 80 48 110.15 48 147.2V208h16c0-16 16-32 32-32h320c16 0 32 16 32 32h16v-60.8c0-37.05-29.69-67.2-66.18-67.2z'/><circle cx='296' cy='232' r='24'/><circle cx='376' cy='232' r='24'/><circle cx='296' cy='312' r='24'/><circle cx='376' cy='312' r='24'/><circle cx='136' cy='312' r='24'/><circle cx='216' cy='312' r='24'/><circle cx='136' cy='392' r='24'/><circle cx='216' cy='392' r='24'/><circle cx='296' cy='392' r='24'/><path stroke-linecap='round' stroke-linejoin='round' d='M128 48v32M384 48v32' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Payout Schedule
        </ion-label>
      </ion-tab-button>
    </ion-tab-bar>
  </div>
</div>
`;

exports[`MainTabs default route should take you to Home Page 1`] = `
<div>
  <div
    class="ion-tabs"
    style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
  >
    <div
      class="tabs-inner"
      style="position: relative; flex: 1; contain: layout size style;"
    >
      <ion-router-outlet>
        <div>
          On the Home Page
        </div>
      </ion-router-outlet>
    </div>
    <ion-tab-bar
      selected-tab="home"
      slot="bottom"
    >
      <ion-tab-button
        href="/"
        tab="home"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M80 212v236a16 16 0 0016 16h96V328a24 24 0 0124-24h80a24 24 0 0124 24v136h96a16 16 0 0016-16V212' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 256L266.89 52c-5-5.28-16.69-5.34-21.78 0L32 256M400 179V64h-48v69' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Home
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/porfolio"
        tab="porfolio"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M384 80H128c-26 0-43 14-48 40L48 272v112a48.14 48.14 0 0048 48h320a48.14 48.14 0 0048-48V272l-32-152c-5-27-23-40-48-40z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M48 272h144M320 272h144M192 272a64 64 0 00128 0M144 144h224M128 208h256' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Porfolio
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/income"
        tab="income"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='32' y='80' width='448' height='256' rx='16' ry='16' transform='rotate(180 256 208)' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M64 384h384M96 432h320' class='ionicon-fill-none ionicon-stroke-width'/><circle cx='256' cy='208' r='80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 160a80 80 0 01-80-80M32 160a80 80 0 0080-80M480 256a80 80 0 00-80 80M32 256a80 80 0 0180 80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Income
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/schedule"
        tab="schedule"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='48' y='80' width='416' height='384' rx='48' ry='48' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M397.82 80H114.18C77.69 80 48 110.15 48 147.2V208h16c0-16 16-32 32-32h320c16 0 32 16 32 32h16v-60.8c0-37.05-29.69-67.2-66.18-67.2z'/><circle cx='296' cy='232' r='24'/><circle cx='376' cy='232' r='24'/><circle cx='296' cy='312' r='24'/><circle cx='376' cy='312' r='24'/><circle cx='136' cy='312' r='24'/><circle cx='216' cy='312' r='24'/><circle cx='136' cy='392' r='24'/><circle cx='216' cy='392' r='24'/><circle cx='296' cy='392' r='24'/><path stroke-linecap='round' stroke-linejoin='round' d='M128 48v32M384 48v32' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Payout Schedule
        </ion-label>
      </ion-tab-button>
    </ion-tab-bar>
  </div>
</div>
`;

exports[`MainTabs default route should take you to home 1`] = `
<div>
  <div
    class="ion-tabs"
    style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
  >
    <div
      class="tabs-inner"
      style="position: relative; flex: 1; contain: layout size style;"
    >
      <ion-router-outlet>
        <div>
          On the Home Page
        </div>
      </ion-router-outlet>
    </div>
    <ion-tab-bar
      selected-tab="home"
      slot="bottom"
    >
      <ion-tab-button
        href="/"
        tab="home"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M80 212v236a16 16 0 0016 16h96V328a24 24 0 0124-24h80a24 24 0 0124 24v136h96a16 16 0 0016-16V212' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 256L266.89 52c-5-5.28-16.69-5.34-21.78 0L32 256M400 179V64h-48v69' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Home
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/porfolio"
        tab="porfolio"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path d='M384 80H128c-26 0-43 14-48 40L48 272v112a48.14 48.14 0 0048 48h320a48.14 48.14 0 0048-48V272l-32-152c-5-27-23-40-48-40z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M48 272h144M320 272h144M192 272a64 64 0 00128 0M144 144h224M128 208h256' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Porfolio
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/income"
        tab="income"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='32' y='80' width='448' height='256' rx='16' ry='16' transform='rotate(180 256 208)' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M64 384h384M96 432h320' class='ionicon-fill-none ionicon-stroke-width'/><circle cx='256' cy='208' r='80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M480 160a80 80 0 01-80-80M32 160a80 80 0 0080-80M480 256a80 80 0 00-80 80M32 256a80 80 0 0180 80' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Income
        </ion-label>
      </ion-tab-button>
      <ion-tab-button
        href="/tabs/schedule"
        tab="schedule"
      >
        <ion-icon
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><rect x='48' y='80' width='416' height='384' rx='48' ry='48' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M397.82 80H114.18C77.69 80 48 110.15 48 147.2V208h16c0-16 16-32 32-32h320c16 0 32 16 32 32h16v-60.8c0-37.05-29.69-67.2-66.18-67.2z'/><circle cx='296' cy='232' r='24'/><circle cx='376' cy='232' r='24'/><circle cx='296' cy='312' r='24'/><circle cx='376' cy='312' r='24'/><circle cx='136' cy='312' r='24'/><circle cx='216' cy='312' r='24'/><circle cx='136' cy='392' r='24'/><circle cx='216' cy='392' r='24'/><circle cx='296' cy='392' r='24'/><path stroke-linecap='round' stroke-linejoin='round' d='M128 48v32M384 48v32' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
        />
        <ion-label>
          Payout Schedule
        </ion-label>
      </ion-tab-button>
    </ion-tab-bar>
  </div>
</div>
`;
